tests/gnmi/test_gnoi_system.py

```diff
@@ -35,6 +35,55 @@ def test_gnoi_system_time(duthosts, rand_one_dut_hostname, localhost):
     pytest_assert("time" in msg_json, "System.Time API did not return time")
 
 
+def test_gnoi_system_reboot(duthosts, rand_one_dut_hostname, localhost):
+    """
+    Verify the gNOI System Reboot API triggers a reboot and the device comes back online.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # Trigger reboot
+    ret, msg = gnoi_request(duthost, localhost, "Reboot", '{"method": 1}')
+    pytest_assert(ret == 0, "System.Reboot API reported failure (rc = {}) with message: {}".format(ret, msg))
+    logging.info("System.Reboot API returned msg: {}".format(msg))
+
+
+def test_gnoi_system_reboot_fail_invalid_method(duthosts, rand_one_dut_hostname, localhost):
+    """
+    Verify the gNOI System Reboot API fails with invalid method.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # Trigger reboot with invalid method
+    ret, msg = gnoi_request(duthost, localhost, "Reboot", '{"method": 2}')
```

Comment:
2 is still a valid method per https://github.com/openconfig/gnoi/blob/0d0e73a9bd74ee5b32b145cb06ac0f613f89ed0c/system/system.proto#L131
cc: @ndas7